includes/ucf-jobs-feed.php

```diff
@@ -0,0 +1,39 @@
+<?php
+/**
+ * Handles all Jobs feed related code.
+ **/
+
+namespace UCF\MainSiteUtilities\Feeds;
+
+/**
+ * Retrieve the job listing data from the given feed url.
+ *
+ * @author Cadie Stockman
+ * @since 2.1.0
+ * @param array $args Arg array
+ * @return object $result Job listings data
+ **/
+function retrieve_job_listing_data( $args ) {
+	$feed_url = get_option( UCF_MAIN_SITE_UTILITIES__CUSTOMIZER_PREFIX . 'jobs_feed_url' );
+	$post_args = array(
+		'body' => wp_json_encode( array(
+			// 'appliedFacets' => {}, // TODO: Add feature for defining job type filters
+			'limit'      => $args['limit'],
+			'offset'     => $args['offset'],
+			'searchText' => ''
+		) ),
+		'headers' => array(
+			'content-type' => 'application/json'
+		)
+	);
+
+	$response      = wp_remote_post( $feed_url, $post_args );
+	$response_code = wp_remote_retrieve_response_code( $response );
+	$result        = false;
+
+	if ( is_array( $response ) && is_int( $response_code ) && $response_code < 400 ) {
+		$result = json_decode( wp_remote_retrieve_body( $response ) );
```

Comment:
This line assumes the values we get back from the endpoint are always valid JSON. If we're comfortable with that, then it's fine, but if we think there's a chance they might send us back junk or malformed JSON, we might want to have some error checking here to it doesn't bork the site.

Reply:
@jmbarne3 - I've added a json check in https://github.com/UCF/Main-Site-Utilities-Plugin/pull/23/commits/83608a394c9901eb6a5514313110a5465f49bb8b. Thoughts on that approach?

Reply:
It just occurred to me that `json_decode` is going to return `null` if the JSON is invalid.. so we really could just handle any sort of error in processing with a null check. My initial concern was really more with making sure we understand what's being returned, and since `json_decode` is going to return null if (1) the body is empty, (2) the `$response` object is an error (which shouldn't happen because of the surrounding checks) or (3) the body of the response is fine, but the JSON is malformed, we can probably just rely on `$result` being null as an indication something went wrong.

That does mean, however, we need to make sure we return _something_ if we just get an empty response. Like if there are no results or whatever the case may be. In that way, we're making null meaningful. Alternatively, if we'd prefer to do a truthy check in our template, we can return false instead of null (kind of like you already are in 83608a3) and that might be simpler/more understandable to test for.

Sorry about the confusion. Totally forgot about `json_decode` returning null on errors.